File: ios/Packages/FeatureServices/WalletService/WalletImportResult.swift
// Copyright (c). Gem Wallet. All rights reserved.

import Primitives

public enum WalletImportResult: Sendable {
    case new(Wallet)
    case existing(Wallet)

    public var wallet: Wallet {
        switch self {
        case .new(let wallet), .existing(let wallet):
            return wallet
        }
    }
}

File: android/gemcore/src/main/kotlin/com/gemwallet/android/cases/wallet/WalletImportResult.kt
package com.gemwallet.android.cases.wallet

import com.wallet.core.primitives.Wallet

sealed class WalletImportResult {
    abstract val wallet: Wallet

    data class New(override val wallet: Wallet) : WalletImportResult()
    data class Existing(override val wallet: Wallet) : WalletImportResult()
}
